src/ast/passes/fold_literals.cpp

```diff
@@ -270,28 +301,28 @@ std::optional<Expression> LiteralFolder::visit(Binop &op)
 
```

Comment:
You are missing the actual handling of `ast::Boolean` in terms of folding. If you have `true && false` then it won't be folded at all AFAICT.

This needs to handle the case of `Boolean [op] anything` (or vice-versa) with the implicit "truthiness" rules that you have defined elsewhere. Then all of this logic can be pushed into the `Cast` operator and handled as an `Unop`.

Reply:
Added handling boolean literals in the folding pass.